aws/src/main/java/org/apache/iceberg/aws/AwsClientProperties.java

```diff
@@ -199,7 +199,9 @@ public AwsCredentialsProvider credentialsProvider(
       String accessKeyId, String secretAccessKey, String sessionToken) {
     if (refreshCredentialsEnabled && !Strings.isNullOrEmpty(refreshCredentialsEndpoint)) {
       clientCredentialsProviderProperties.put(
-          VendedCredentialsProvider.URI, refreshCredentialsEndpoint);
+          VendedCredentialsProvider.CREDENTIALS_ENDPOINT, refreshCredentialsEndpoint);
+      clientCredentialsProviderProperties.put(
+          VendedCredentialsProvider.URI, allProperties.get(CatalogProperties.URI));
```

Comment:
rather than introducing/renaming the existing property, we could just do 
```
      Optional.ofNullable(allProperties.get(allProperties.get(CatalogProperties.URI)))
          .ifPresent(
              catalogUri ->
                  clientCredentialsProviderProperties.put(CatalogProperties.URI, catalogUri));
```